### PR TITLE
CFM-1192: Create view block to display suggested groups

### PR DESCRIPTION
**Pull request by DavidHernandez**
Updates the suggested groups block to display 5 items, order them alphabetically and send the user interests as default filters on the group page.

Issue: https://issuetracker.amplexor.com/jira/browse/CFM-1192

suggested-groups

### CODE REVIEW

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/c4m_features_homepage.module [hunk 1]
```diff
@@ -54,30 +54,54 @@ function c4m_features_homepage_c4m_suggested_groups_get_groups() {
...
+    // groups page.
+    foreach ($interests as $key => $interest) {
+      $link_filters['f[' . $key . ']'] = 'c4m_vocab_topic:' . $interest;
+    }
   }
 
   $result = $query
     ->addTag('random')
-    ->range(0, 3)
+    ->range(0, 5)
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/c4m_features_homepage.module]: @DavidHernandez 
why `5` ?
can we use constant ?

It seems that somewhere else we need 3 ...

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/c4m_features_homepage.module [hunk 2]
```diff
@@ -54,30 +54,54 @@ function c4m_features_homepage_c4m_suggested_groups_get_groups() {
...
     return;
   }
 
+  // After selecting 5 random groups, they are ordered alphabetically.
+  $nodes = array();
   foreach ($result['node'] as $group) {
-    $node_view = node_view(node_load($group->nid), 'block_list');
+    $nodes[] = node_load($group->nid);
+  }
+  uasort($nodes, 'c4m_features_homepage_node_sort_by_title');
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/c4m_features_homepage.module]: @DavidHernandez 

isn't it better to sort it on the query itself ?

Comment (DavidHernandez) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/c4m_features_homepage.module]: The query is already sorted as random. The MySQL sort acts before the query limit. In our case we need to sort first as random, limit the amount of items, sort the selected items alphabetically. This is not possible to do it in MySQL without a subquery or with two selects. In this case, I prefer to do the sort on the PHP side.

On the other hand, we could drop the random sort, but then, we will always return the same items.

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/c4m_features_homepage.module (hunk 1 above)
Comment (DavidHernandez) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/c4m_features_homepage.module]: As far as I know, this pane is used only on the homepage. I could only find one occurrence in the code.

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/c4m_features_homepage.module [hunk 3]
```diff
@@ -54,30 +54,54 @@ function c4m_features_homepage_c4m_suggested_groups_get_groups() {
...
 
+  $link_filters = array();
   if (!empty($interests)) {
     $query->fieldCondition('c4m_vocab_topic', 'tid', $interests, 'IN');
+
+    // Adds each interest to the exposed filters that will be pass to the
+    // groups page.
+    foreach ($interests as $key => $interest) {
+      $link_filters['f[' . $key . ']'] = 'c4m_vocab_topic:' . $interest;
+    }
```
Comment (soniCaH) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/c4m_features_homepage.module]: Isn't it easier to link to `/groups?filter=interests`?

**Pull request merged**